src/ophyd_async/epics/pmac/_pmac_trajectory.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+from dataclasses import dataclass
+
+import numpy as np
+from scanspec.core import Path
+from scanspec.specs import Spec
+
+from ophyd_async.core import FlyerController, set_and_wait_for_value, wait_for_value
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.pmac import PmacIO
+from ophyd_async.epics.pmac._pmac_io import CS_LETTERS
+from ophyd_async.epics.pmac._utils import (
+    TICK_S,
+    _PmacMotorInfo,
+    _Trajectory,
+    calculate_ramp_position_and_duration,
+)
+
+
+@dataclass
+class PmacTriggerLogic:
+    spec: Spec[Motor]
+
+
+class PmacTrajectoryTriggerLogic(FlyerController[PmacTriggerLogic]):
+    def __init__(self, pmac: PmacIO) -> None:
+        self.pmac = pmac
+        self.scantime: float | None = None
+
+    async def prepare(self, value: PmacTriggerLogic):
+        _slice = Path(value.spec.calculate()).consume()
```

Comment:
Nit: local variables are always private, so don't need the leading undescore
```suggestion
        slice = Path(value.spec.calculate()).consume()
```